### ISSUE TITLE
Ability to choose the latest minclientversion automatically.

### ISSUE THREAD

**Issue by YelehaUwU** [multitheftauto/mtasa-blue#3577]
### Describe the solution you'd like

I would like the value **latest** to be implemented as a parameter of **minclientversion** in the **mtaserver.conf** and preferably set as default, for servers to be up to date and make the server config require less attention.

### Security Policy

- [X] I have read and understood the Security Policy and this issue is not about a cheat or security vulnerability.

**Comment by TracerDS** [comment]
Mta does version checks when client joins. It checks against its version, not against the latest one.
Server would have to sync with mta's nightly to get the latest version number.
That would be quite a hassle.

In my opinion this is pointless and it just introduces laziness.

**Comment by YelehaUwU** [comment]
https://github.com/multitheftauto/mtasa-blue/blob/b6fc1b5156337d0ad5be540881b71b8ee362fd3a/Server/mods/deathmatch/utils/CHqComms.h#L147

Part of it's already implemented here as part of HQ updates, how would it be a hassle to implement?

There could be a setting that would determine if it should automatically change to latest and just reuse this function.

**Comment by TracerDS** [comment]
> https://github.com/multitheftauto/mtasa-blue/blob/b6fc1b5156337d0ad5be540881b71b8ee362fd3a/Server/mods/deathmatch/utils/CHqComms.h#L147
> 
> Part of it's already implemented here as part of HQ updates, how would it be a hassle to implement?
> 
> There could be a setting that would determine if it should automatically change to latest and just reuse this function.

it processes it from the client, hence the `bitStream`.
It would be a hassle because MTA would need to create an API for just returning latest version, update it every build and sync it with every server on the planet.
Its trivial, but it takes time with little to no benefit at all.

**Comment by YelehaUwU** [comment]
Can't we use https://nightly.multitheftauto.com/ver/ data?

**Comment by TracerDS** [comment]
> Can't we use https://nightly.multitheftauto.com/ver/ data?

Yes, we can use it to fetch current version **if** there is a way to get pure version info for current release (1.6).
If not, there would be a lot of fiddling with webscraping

**Comment by Fernando-A-Rocha** [comment]
Good idea. Just add an endpoint like https://nightly.multitheftauto.com/ver/latest that returns the latest nightly version string meant to be used by all servers

**Comment by jlillis** [comment]
This is already possible using [`minclientversion_auto_update`](https://wiki.multitheftauto.com/wiki/Server_mtaserver.conf#minclientversion_auto_update). You can also use the `acpanel` resource to set this and many more anticheat related options.

> Good idea. Just add an endpoint like https://nightly.multitheftauto.com/ver/latest that returns the latest nightly version string meant to be used by all servers

This already exists. [An example of it in use is available in the `acpanel` resource.](https://github.com/multitheftauto/mtasa-resources/blob/3dd5cbd32f092337707277fbecc5ee54988e07fc/%5Badmin%5D/acpanel/s_settings.lua#L197)

**Comment by YelehaUwU** [comment]
The setting [minclientversion_auto_update](https://wiki.multitheftauto.com/wiki/Server_mtaserver.conf#minclientversion_auto_update) doesn't work for some reason.
I'll see about tweaking the acpanel resource as it gets the 1.3 version instead of 1.6

**Comment by TracerDS** [comment]
> The setting [minclientversion_auto_update](https://wiki.multitheftauto.com/wiki/Server_mtaserver.conf#minclientversion_auto_update) doesn't work for some reason. I'll see about tweaking the acpanel resource as it gets the 1.3 version instead of 1.6

Hmmm, it seems its not used anywhere. Just defined but not used
![image](https://github.com/user-attachments/assets/3402e586-62a7-4766-92cc-13ee91ffa0e2)

**Comment by Fernando-A-Rocha** [comment]
@Dutchman101 can you elaborate on the purpose of minclientversion_auto_update, considering you are in the MTA AC team?

**Comment by Xenius97** [comment]
U can do it with a simple script, just using lua:

```lua
function refreshMinVersion()
	fetchRemote("https://nightly.multitheftauto.com/ver/", function(body, err)
		local minclientVersion = "1.6.0-9.22628.0"
		if err == 0 then
			local version = "1.6"
			local pattern = "<b>version: " .. version .. "</b>.-Max recommended/minclientversion:%s*([%d%.%-]+)"
			
			minclientVersion = string.match(body, pattern)
		else
			iprint("Error fetching data: " .. err)
		end

		setServerConfigSetting("minclientversion", minclientVersion, true) -- https://nightly.mtasa.com/ver/ -> Max recommended/minclientversion
		iprint("Min client version set to: " .. minclientVersion)
	end)
end

addEventHandler("onResourceStart", resourceRoot, function()
	refreshMinVersion()
	setTimer(refreshMinVersion, 60000 * 60 * 8, 0) -- optional [auto update every 8 hour]
end)
```

**Comment by YelehaUwU** [comment]
It's nice, but we're trying to find out the reasoning behind it not being implemented in the server config in a functional way when the setting exists.

**Comment by Fernando-A-Rocha** [comment]
> It's nice, but we're trying to find out the reasoning behind it not being implemented in the server config in a functional way when the setting exists.

I just tested something. I set `minclientversion` in mtaserver.conf to empty, and `minclientversion_auto_update` to `2`. After starting the server, this setting was set to `1.6.0-9.22628.0` automatically. So this feature is doing something, it's just not very transparent on what happens...

**Comment by YelehaUwU** [comment]
Can you try doing the same while the server is turned off? Maybe the file was in use and thus got rewritten back, happened to me few times.

**Comment by Fernando-A-Rocha** [comment]
What you mean exactly?

**Comment by YelehaUwU** [comment]
Maybe the server config wasnt properly saved due to the server running

-------- Pôvodná správa --------
24. 7. 2024 15:09, Fernando Rocha  napísal/a:

**Comment by Fernando-A-Rocha** [comment]
Server doesn't detect config changes if you edit mtaserver.conf while the server is running. It reads the values on server startup

**Comment by YelehaUwU** [comment]
Whenever I edited it while it was running, there was a 50/50 chance it would revert the changes on restart :d

-------- Pôvodná správa --------
24. 7. 2024 16:14, Fernando Rocha  napísal/a:
